src/components/notifications/WebhookConfigModal.tsx

```diff
@@ -0,0 +1,386 @@
+        this.renderHeadersList = this.renderHeadersList.bind(this);
+        this.setCopied = this.setCopied.bind(this);
+        this.copyToClipboard = this.copyToClipboard.bind(this);
+        this.isValid = this.isValid.bind(this);
+    }
+
+    componentDidMount() {
+        if (this.props.webhookConfigId) {
+            getWebhookConfiguration(this.props.webhookConfigId)
+                .then((response) => {
+                    let state = { ...this.state };
+                    const _headers = [...this.state.form.header]
+                    state.view = ViewType.FORM;
+                    const _responseKeys = response.result?.header ? Object.keys(response.result.header) : undefined
+                    {_responseKeys && _responseKeys.map((_key) => {
+                        _headers.push({ key: _key, value: response.result?.header[_key] })
+                    })}
+                    const _responsePayload = response.result?.payload ? JSON.stringify(response.result?.payload) : ""
+                    state.form = {
+                        ...response.result,
+                        header: _headers,
+                        payload: _responsePayload    
+                    };
+                    state.isValid = {
+                        configName: true,
+                        webhookUrl: true,
+                        payload: true,
+                    }
+                    this.setState(state);
+                }).catch((error) => {
+                    showError(error);
+                })
+        }
+        else {
+            getProjectListMin().then((response) => {
+                this.setState({
+                    view: ViewType.FORM
+                })
+            }).catch((error) => {
+                showError(error);
+            })
+        }
+        getWebhookAttributes()
+            .then((response) => {
+                let state = { ...this.state }
+                state.webhookAttribute = { ...response.result };
+                this.setState(state)
+            }).catch((error) => {
+                showError(error)
+            })
+    }
+
+    handleWebhookConfigNameChange(event: React.ChangeEvent<HTMLInputElement>,): void {
+        let { form } = { ...this.state };
+        form.configName = event.target.value;
+        this.setState({ form });
+    }
+
+    isValid(event, key: 'configName' | 'webhookUrl' | 'payload'): void {
+        let { form, isValid } = { ...this.state };
+        if(key!='payload'){
+            isValid[key] = event.target.value.length !== 0;
+        }else if(this.state.form.payload != '') {
+            try{
+                JSON.parse(this.state.form.payload)
+                isValid[key] = true
+            }catch(err) {
+                isValid[key] = false
+            }
+        }
+        this.setState({ form, isValid });
+    }
+
+    handleWebhookUrlChange(event: React.ChangeEvent<HTMLInputElement>): void {
+        let { form } = { ...this.state };
```

Comment:
use const where let is used & not reassigning the values

Reply:
here value is reassigning , checked it with const, not working